pkg/cmd/config/set_garden.go

```diff
@@ -0,0 +1,166 @@
+	// Name is a unique name of this Garden that can be used to target this Garden
+	Name string
+	// KubeconfigFlag is the path to the kubeconfig file of the Garden cluster
+	KubeconfigFlag flag.StringFlag
+	// ContextFlag Overrides the current-context of the garden cluster kubeconfig
+	// +optional
+	ContextFlag flag.StringFlag
+	// Patterns is a list of regex patterns that can be defined to use custom input formats for targeting
+	// Use named capturing groups to match target values.
+	// Supported capturing groups: project, namespace, shoot
+	// +optional
+	Patterns []string
+}
+
+// Complete adapts from the command line args to the data required.
+func (o *setGardenOptions) Complete(f util.Factory, cmd *cobra.Command, args []string) error {
+	config, err := getConfiguration(f)
+	if err != nil {
+		return err
+	}
+
+	o.Configuration = config
+
+	if len(args) > 0 {
+		o.Name = strings.TrimSpace(args[0])
+	}
+
+	return nil
+}
+
+// Validate validates the provided options
+func (o *setGardenOptions) Validate() error {
+	if o.Name == "" {
+		return errors.New("garden identity is required")
+	}
+
+	if err := validatePatterns(o.Patterns); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// AddFlags adds flags to adjust the output to a cobra command
+func (o *setGardenOptions) AddFlags(flags *pflag.FlagSet) {
+	flags.Var(&o.KubeconfigFlag, "kubeconfig", "path to kubeconfig file for this Garden cluster")
+	flags.Var(&o.ContextFlag, "context", "override the current-context of the garden cluster kubeconfig")
+	flags.StringArrayVar(&o.Patterns, "pattern", nil, `define regex match patterns for this garden for custom input formats for targeting.
+Use named capturing groups to match target values.
+Supported capturing groups: project, namespace, shoot.
+Note that if you set this flag it will overwrite the pattern list in the config file.
+You may specify any number of extra patterns.
+Example: ^((?Pmy-garden[^/]+)/)?shoot--(?P<project>.+)--(?P<shoot>.+)$`)
+}
+
+// Run executes the command
+func (o *setGardenOptions) Run(_ util.Factory) error {
+	garden, err := o.Configuration.Garden(o.Name)
+	if err == nil {
+		if o.KubeconfigFlag.Provided() {
+			garden.Kubeconfig = o.KubeconfigFlag.Value()
+		}
+
+		if o.ContextFlag.Provided() {
+			garden.Context = o.ContextFlag.Value()
+		}
+
+		if o.Patterns != nil {
+			firstPattern := o.Patterns[0]
+			if len(firstPattern) > 0 {
+				garden.Patterns = o.Patterns
+			} else {
+				garden.Patterns = []string{}
+			}
+		}
+	} else {
+		o.Configuration.Gardens = append(o.Configuration.Gardens, config.Garden{
+			Name:       o.Name,
+			Kubeconfig: o.KubeconfigFlag.Value(),
+			Context:    o.ContextFlag.Value(),
+			Patterns:   o.Patterns,
+		})
+	}
+
+	err = o.Configuration.Save()
+	if err != nil {
+		return fmt.Errorf("failed to configure garden: %w", err)
+	}
+
+	fmt.Fprintf(o.IOStreams.Out, "Successfully configured garden %q\n", o.Name)
+
+	return nil
+}
+
+func validatePatterns(patterns []string) error {
```

Comment:
How should this flag work in detail. Should it be possible to delete a single or all patterns?